base/collection_view.go

```diff
@@ -379,3 +380,9 @@ func asViewConsistency(value interface{}) gocb.ViewScanConsistency {
 	}
 
 }
+
+// GetGoCBAgent returns the underlying agent from gocbcore
+func (c *Collection) GetGoCBAgent() (*gocbcore.Agent, error) {
```

Comment:
I think it would be preferable to put the functionality that uses the agent on the Collection API (i.e. add getCollectionID here), to avoid leaking dependencies on the underlying gocb implementation.

Reply:
Moved code here, but it'll get moved to a database level call when we run on multiple collections.